modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java

```diff
@@ -423,4 +423,11 @@ public interface MetaStorageManager extends IgniteComponent {
 
     /** Unregisters a Meta Storage revision update listener. */
     void unregisterRevisionUpdateListener(RevisionUpdateListener listener);
+
+    /**
+     * Returns the local compaction revision that was set or restored from a metastorage snapshot, {@code -1} if not changed.
+     *
+     * @throws IgniteInternalException with cause {@link NodeStoppingException} if the node is in the process of stopping.
```

Comment:
That's what `org.apache.ignite.internal.util.IgniteUtils#inBusyLock(org.apache.ignite.internal.util.IgniteSpinBusyLock, java.util.function.IntSupplier)` does.

Reply:
Why do we use this method then? What's the point of this wrapping?

Reply:
This is a convenient auxiliary method, which is already used in many parts when, it can be remade for all, but not in this task of course. I think that the main headache is that the exception extends the checked exception, which of course we do not like very much now.
NodeStoppingException extends IgniteInternalCheckedException